pages/events/[eventsId].js

```diff
@@ -0,0 +1,142 @@
+import { styled } from "styled-components";
+import { useRouter } from "next/router";
+import { events } from "@/api/db";
+import { useEffect, useState } from "react";
+
+export default function EventsDetail({ handleUpdate }) {
+  const [editMode, setEditMode] = useState(false);
+  const [eventObject, setEventObject] = useState(null);
+
+  const router = useRouter();
+  const eventsId = router.query.eventsId;
+
+  const eventToEdit = events.find((event) => event.id == eventsId);
+
+  useEffect(() => {
+    setEventObject(eventToEdit);
+  }, [eventToEdit]);
+
+  const handleEditClick = () => {
+    setEditMode(!editMode);
+  };
+  const handleInputChange = (e) => {
+    const { name, value } = e.target;
+    setEventObject((prevEvent) => ({
+      ...prevEvent,
+      [name]: value,
+    }));
+  };
+
+  const handleSaveClick = () => {
+    setEditMode(false);
+    handleUpdate(eventObject);
+  };
+
+  if (!eventToEdit) {
+    return;
+  }
+
+  return (
+    <article>
+      <TitleContainer>
+        {editMode ? (
+          <input
+            type="text"
+            name="title"
+            value={eventObject.title}
+            onChange={handleInputChange}
+          />
+        ) : (
+          eventObject?.title
+        )}
+      </TitleContainer>
+      <DateContainer>
+        {editMode ? (
+          <input
+            type="text"
+            name="date"
+            value={eventObject?.date}
+            onChange={handleInputChange}
+          />
+        ) : (
+          eventObject?.date
+        )}
+      </DateContainer>
+      <TypeContainer>
+        {editMode ? (
+          <input
+            type="text"
+            name="type"
+            value={eventObject?.type}
+            onChange={handleInputChange}
+          />
+        ) : (
+          eventObject?.type
+        )}
+      </TypeContainer>
+      <DescriptionContainer>
+        {editMode ? (
+          <textarea
+            name="description"
+            value={eventObject?.description}
+            onChange={handleInputChange}
+          />
+        ) : (
+          eventObject?.description
+        )}
+      </DescriptionContainer>
+      <ButtonContainer>
+        {editMode ? (
+          <button onClick={handleSaveClick}>Save Changes</button>
+        ) : (
+          <button onClick={handleEditClick}>EDIT EVENT</button>
+        )}
+      </ButtonContainer>
+    </article>
```

Comment:
instead of asking 5 times if you are in edit mode, you can also use an if statement with an "early return":
```
if (!eventToEdit) {
    return;
  }

if(editMode) {
  return  <article>
    ...the article in edit mode with inputs
    </article>
}

return  <article>
    ...the article in normal mode, with the info displayed.
    </article>
```

Reply:
@F-Kirchhoff could you please clarify this for me?

Reply:
I try to clarify it for you: Felix wanted to say that you can reduce the amount of conditional renderings via an early return. This is also what Roland tried to suggest to you above. You are always asking for the same condition (is editMode true or false). Your code is easier to read if you have multiple returns in your component and one is nested inside an if statement.

```jsx

if (editMode) {
  return (<article>
      <TitleContainer>
          <input
            type="text"
            name="title"
            value={eventObject.title}
            onChange={handleInputChange}
          />
      </TitleContainer>
      <DateContainer>
          <input
            type="text"
            name="date"
            value={eventObject?.date}
            onChange={handleInputChange}
          />
      </DateContainer>
      <TypeContainer>
          <input
            type="text"
            name="type"
            value={eventObject?.type}
            onChange={handleInputChange}
          />
      </TypeContainer>
      <DescriptionContainer>
          <textarea
            name="description"
            value={eventObject?.description}
            onChange={handleInputChange}
          />
      </DescriptionContainer>
      <ButtonContainer>
          <button onClick={handleSaveClick}>Save Changes</button>
      </ButtonContainer>
    </article>)
}

return (
  <article>
      <TitleContainer>
          eventObject?.title
      </TitleContainer>
      <DateContainer>
          eventObject?.date
      </DateContainer>
      <TypeContainer>
          eventObject?.type
      </TypeContainer>
      <DescriptionContainer>
          eventObject?.description
      </DescriptionContainer>
      <ButtonContainer>
          <button onClick={handleEditClick}>EDIT EVENT</button>
      </ButtonContainer>
    </article>)

```

I hope this does help to clarify this for you more. :)